packages/bsky/src/services/actor/views.ts

```diff
@@ -95,79 +85,100 @@ export class ActorViews {
-      this.getListMutes(dids, viewer),
+      this.services.label.getLabelsForSubjects(skipLabels ? [] : dids),
     ])
 
-    const profileInfoByDid = profileInfos.reduce((acc, info) => {
-      return Object.assign(acc, { [info.did]: info })
-    }, {} as Record<string, ArrayEl<typeof profileInfos>>)
+    const listUris: string[] = profileInfos
+      .map((a) => a.requesterMutedByList)
+      .filter((list) => !!list)
+    const listViews = await this.services.graph.getListViews(listUris, viewer)
 
-    const views = results.map((result) => {
-      const profileInfo = profileInfoByDid[result.did]
-      const avatar = profileInfo?.avatarCid
+    return profileInfos.reduce((acc, cur) => {
+      const avatar = cur?.avatarCid
         ? this.imgUriBuilder.getCommonSignedUri(
             'avatar',
-            profileInfo.did,
-            profileInfo.avatarCid,
+            cur.did,
+            cur.avatarCid,
           )
         : undefined
-      const banner = profileInfo?.bannerCid
+      const banner = cur?.bannerCid
         ? this.imgUriBuilder.getCommonSignedUri(
             'banner',
-            profileInfo.did,
-            profileInfo.bannerCid,
+            cur.did,
+            cur.bannerCid,
           )
         : undefined
-      return {
-        did: result.did,
-        handle: result.handle ?? INVALID_HANDLE,
-        displayName: profileInfo?.displayName || undefined,
-        description: profileInfo?.description || undefined,
+      const profile = {
+        did: cur.did,
+        handle: cur.handle ?? INVALID_HANDLE,
+        displayName: cur?.displayName || undefined,
+        description: cur?.description || undefined,
         avatar,
         banner,
-        followsCount: profileInfo?.followsCount ?? 0,
-        followersCount: profileInfo?.followersCount ?? 0,
-        postsCount: profileInfo?.postsCount ?? 0,
-        indexedAt: profileInfo?.indexedAt || undefined,
+        followsCount: cur?.followsCount ?? 0,
+        followersCount: cur?.followersCount ?? 0,
+        postsCount: cur?.postsCount ?? 0,
+        indexedAt: cur?.indexedAt || undefined,
         viewer: viewer
           ? {
-              following: profileInfo?.requesterFollowing || undefined,
-              followedBy: profileInfo?.requesterFollowedBy || undefined,
-              muted: !!profileInfo?.requesterMuted || !!listMutes[result.did],
-              mutedByList: listMutes[result.did],
-              blockedBy: !!profileInfo.requesterBlockedBy,
-              blocking: profileInfo.requesterBlocking || undefined,
+              following: cur?.requesterFollowing || undefined,
+              followedBy: cur?.requesterFollowedBy || undefined,
+              muted: !!cur?.requesterMuted || !!cur.requesterMutedByList,
+              mutedByList: cur.requesterMutedByList
+                ? this.services.graph.formatListViewBasic(
+                    listViews[cur.requesterMutedByList],
```

Comment:
It's technically possible this list view might not exist to pass for formatting.